youtube_dl/extractor/animelab.py

```diff
@@ -0,0 +1,231 @@
+    urlencode_postdata,
+    int_or_none,
+    str_or_none,
+)
+
+from ..compat import compat_HTTPError
+
+
+class AnimeLabBaseIE(InfoExtractor):
+    _LOGIN_REQUIRED = True
+    _LOGIN_URL = 'https://www.animelab.com/login'
+    _NETRC_MACHINE = 'animelab'
+
+    def _login(self):
+        def is_logged_in(login_webpage):
+            return '<title>AnimeLab - Login' not in login_webpage
+
+        login_page = self._download_webpage(
+            self._LOGIN_URL, None, 'Downloading login page')
+
+        # Check if already logged in
+        if is_logged_in(login_page):
+            return
+
+        (username, password) = self._get_login_info()
+        if username is None and self._LOGIN_REQUIRED:
+            self.raise_login_required('Login is required to access any AnimeLab content')
+
+        login_form = {
+            'email': username,
+            'password': password,
+        }
+
+        try:
+            response = self._download_webpage(
+                self._LOGIN_URL, None, 'Logging in', 'Wrong login info',
+                data=urlencode_postdata(login_form),
+                headers={'Content-Type': 'application/x-www-form-urlencoded'})
+        except ExtractorError as e:
+            if isinstance(e.cause, compat_HTTPError) and e.cause.code == 400:
+                page = e.cause.read().decode('utf-8')
+                js_error = self._search_regex(
+                    r'Utils\.notify\( *?[\'"]error[\'"] *?, *?[\'"](.*?)[\'"]',
+                    page, 'Trying to get error message in js', default=None, fatal=False)
+                if js_error:
+                    raise ExtractorError('Unable to log in: %s' % js_error, expected=True)
+
+                raise ExtractorError('Unable to log in (could not find error)', expected=True)
+            else:
+                raise
+
+        # if login was successful
+        if is_logged_in(response):
+            return
+
+        raise ExtractorError('Unable to login (cannot verify if logged in)')
+
+    def _real_initialize(self):
+        self._login()
+
+    def _extract_video_collection_from_player(self, webpage):
+        video_collection_str = self._search_regex(r'new +?VideoCollection *?\((.*?)\);', webpage, 'AnimeLab VideoCollection')
+        return self._parse_json(video_collection_str, None)
+
+    def _extract_position_from_player(self, webpage):
+        position_str = self._search_regex(r'playlistPosition *?= *?(\d+)', webpage, 'Playlist Position')
+        position = int_or_none(position_str)
+        if position is None:
+            raise ExtractorError('Could not get playlist position as integer. Extracted string was %s' % position_str)
+
+        return position
+
+    def _create_title(self, webpage, title_parts):
+        if None not in title_parts:
+            return '%s - %s %s - %s' % title_parts
+        else:
+            self.report_warning('Could not get all necessary data to contruct title manually, trying alternative sources...')
```

Comment:
So I should rely on title from HTML alone (I'm just not sure exactly what to remove)?